### PR TITLE
integer64 support (follow up to #121)

### PR DESCRIPTION
**Pull request by romainfrancois**

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -135,6 +135,7 @@ S3method(vec_arith,POSIXct)
 S3method(vec_arith,default)
 S3method(vec_arith,difftime)
 S3method(vec_arith,factor)
+S3method(vec_arith,integer)
```
Comment (romainfrancois) [inline NAMESPACE]: @Hadley should this be eliminated too ?

File: R/arith.R [hunk 2]
```diff
@@ -63,6 +63,9 @@ vec_arith.logical.logical <- function(op, x, y) vec_arith_base(op, x, y)
 #' @method vec_arith.logical numeric
 #' @export
 vec_arith.logical.numeric <- function(op, x, y) vec_arith_base(op, x, y)
+#' @method vec_arith.logical integer
+#' @export
+vec_arith.logical.integer <- function(op, x, y) vec_arith_base(op, x, y)
```
Comment (hadley) [inline R/arith.R]: Are you sure this is necessary?

Comment (romainfrancois) [reply R/arith.R]: Removed. I added it for symmetry when I needed `vec_arith.integer` for `integer64`. Obsolete now.

**Comment by romainfrancois** [comment]
Is it worth doing this as a record with a single field for better control of things and then relay back to bit64 only for the low level ?

**Comment by hadley** [comment]
I'd prefer to stick with raw bit64 for now

File: DESCRIPTION [hunk 3]
```diff
@@ -27,7 +27,8 @@ Suggests:
     pkgdown,
     rmarkdown,
     testthat,
-    tibble
+    tibble,
+    bit64
```
Comment (javierluraschi) [inline DESCRIPTION]: Can we really just add `bit64` as suggest and not import? In `arrow`, it was easy to hit crashes if `bit64` gets unloaded, see https://github.com/apache/arrow/pull/2867, but maybe that's not the case in `vctrs`, just passing along FWIW.

Comment (romainfrancois) [reply DESCRIPTION]: I think it's fine to use `Suggests` and then `bit64::` when we need a function from it.

**Comment by romainfrancois** [comment]
Was there anything else needed for this ?

**Comment by hadley** [comment]
I just need some time to think it through. Is it blocking you?

**Comment by romainfrancois** [comment]
Not really, `arrow` just uses a remote, that's fine. 

https://github.com/apache/arrow/blob/master/r/DESCRIPTION#L31

File: R/type-integer64.R [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+#' @rdname new_int64
+#' @export
+vec_ptype_abbr.integer64 <- function(x) {
+  "int64"
+}
+
+
+# Coerce ------------------------------------------------------------------
+
+#' @export
```
Comment (hadley) [inline R/type-integer64.R]: Can you please check this is up to date with the advice in https://vctrs.r-lib.org/articles/s3-vector.html#double-dispatch?

Comment (romainfrancois) [reply R/type-integer64.R]: I'm just unsure about 

```r
#' @export
#' @method vec_cast.integer64 logical
vec_cast.integer64.logical <- function(x, to) {
  bit64::as.integer64(x)
}
```

Does this need to distinguish the unspecified case given that cast to logical is allowed, or is this good enough: 

``` r
library(vctrs)
x <- bit64::as.integer64(1:10)
vec_cast(x, unspecified())
#> Error: Can't cast <integer64> to <vctrs_unspecified>
vec_cast(x, NA)
#>  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
```

<sup>Created on 2018-11-20 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

Comment (hadley) [reply R/type-integer64.R]: It's the other way around:

```R
library(vctrs)
x <- bit64::as.integer64(1:10)
vec_cast(unspecified(), x)
vec_cast(NA, x)
```

And I think you need to use `vec_unspecified_cast(x, to)` in that method

**Comment by hadley** [comment]
Looks good. I'll merge once vctrs is on CRAN.

**Comment by javierluraschi** [comment]
@hadley can we merge since `vctrs` is in CRAN?

**Comment by hadley** [comment]
Done!

**Pull request merged**